src/app_engine/compute_page.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+
+"""Compute page for handling tasks related to compute engine."""
+
+import logging
+
+import webapp2
+
+import apiauth
+from google.appengine.api import app_identity
+from google.appengine.api import taskqueue
+
+
+# Compute page constants
+ACTION_START = 'start'
+ACTION_STATUS = 'status'
+
+# Constants for the Compute Engine API
+COMPUTE_STATUS = 'status'
+COMPUTE_STATUS_TERMINATED = 'TERMINATED'
+COMPUTE_STATUS_RUNNING = 'RUNNING'
+
+# Seconds between API retries.
+TASK_RETRY_S = 3
+
+COMPUTE_API_URL = 'https://www.googleapis.com/auth/compute'
+
+
+class ComputePage(webapp2.RequestHandler):
+  """Page to handle requests against GCE."""
+
+  compute_service = None
+
+  def __init__(self, request, response):
+    self.initialize(request, response)
+    if ComputePage.compute_service is None:
+      ComputePage.compute_service = self._build_compute_service()
+    if ComputePage.compute_service is None:
+      logging.warning('Unable to create Compute service object.')
+
+  def _build_compute_service(self):
+    return apiauth.build(scope=COMPUTE_API_URL,
+                         service_name='compute',
+                         version='v1')
+
+  def _enqueue_start_task(self, instance, zone):
+    taskqueue.add(url='/compute/start/%s/%s' % (instance, zone),
+                  countdown=TASK_RETRY_S)
+
+  def _maybe_start_instance(self, instance, zone):
+    """Start the compute instance if it is not already started.
+
+    Args:
+      instance: Name of the instance to start.
+      zone: Name of the zone the instance belongs to.
+    """
+    if ComputePage.compute_service is None:
+      logging.warning('Unable to start Compute instance, service unavailable.')
+      return
+
+    status = self._compute_status(instance, zone)
+
+    logging.info('GCE VM \'%s (%s)\' status: \'%s\'.',
+                 instance, zone, status)
+
+    if status == COMPUTE_STATUS_RUNNING:
+      return
+
+    if status == COMPUTE_STATUS_TERMINATED:
+      logging.info('Starting GCE VM: %s (%s)', instance, zone)
+      ComputePage.compute_service.instances().start(
+          project=app_identity.get_application_id(),
+          instance=instance,
+          zone=zone).execute()
```

Comment:
because we want to make sure that the task starts so we want to queue the task again. i'm actually not sure how this is going to work in practice. I may need to actually name the task in the case where the prober fails again and fires off another task.....

hrm, need to think about this more.